tests/unit/proxy/test_elasticsearch.py

```diff
@@ -320,6 +320,133 @@ def test_search_regex_match_field(self,
                              vars(expected.results[0]),
                              "Search result doesn't match with expected result!")
 
+    @patch('elasticsearch_dsl.Search.execute')
+    def test_search_table_filter(self, mock_search: MagicMock) -> None:
+        mock_results = MagicMock()
+        mock_results.hits.total = 1
+        mock_results.__iter__.return_value = [Response(result=vars(self.mock_result1))]
+        mock_search.return_value = mock_results
```

Comment:
what if you didn't mock the search result and let the test actually hit elasticsearch (some sort of round trip test - load a test document into the index, then search on it). I'd be curious to do that test, especially with these huge changes. This test is good for shared proxy testing, but not great for understanding if the changes are actually doing what we expect under the hood

Reply:
I get what you're saying -- this particular test verifies that the method returns what we mocked it to return, not that it interacted with the elasticsearch index in the way that we want it to interact with it. I'm open a contribution that will make that possible by introducing as test document, and then appropriately modifying some of these tests.

Reply:
Created an issue: https://github.com/lyft/amundsen/issues/280